### PR TITLE
Auto power loss print resume with temperature limitation

### LINKED ISSUE #26528: [FR] Auto power loss print resume with temperature limitation
### Is your feature request related to a problem? Please describe.

Currently after power loss and restart printer sometimes ask confirmation for resume (M1000 S), and sometimes continue automatically (M1000) - https://github.com/search?q=repo%3AMarlinFirmware%2FMarlin+%22M1000&type=code It's hard coded for different cases (screen types?).

Confirmation approach could cause resume delay, until bed is cooled too much so model can not be further printed due adhesion loss.
Auto approach could resume printing when bed is cool and it's already too late to resume, thus causing double loss of filament.

For instance ASA on glass bed almost losses adhesion under 90-80 deg.C.

### Are you looking for hardware support?

_No response_

### Describe the feature you want

I would suggest to add resume temperature threshold parameter, with meaning to resume printing if temperature if more than threshold, and cancel (or ask) if it's already cooler.

Logic example is something like this:

```
void GcodeSuite::M1000() {

  if (recovery.valid()) {
     bool force_resume    = false;

     if (thermalManager.degBed() >= 90)
        force_resume = true;

    if (parser.seen_test('S') && !force_resume) {
```

Setting threshold to something high (or -1, or adding additional flag) will trigger current behavior. Such parameters could be define in config_adv.h, or dynamic EEPROM parameters.

If such change is welcome I could work on it. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vovodroid**
Closes #26528 

Currently after power loss and restart printer sometimes ask confirmation for resume (M1000 S), and sometimes continue automatically (M1000) - https://github.com/search?q=repo%3AMarlinFirmware%2FMarlin+%22M1000&type=code It's hard coded for different cases (screen types?).

Confirmation approach could cause resume delay, until bed is cooled too much so model can not be further printed due adhesion loss.
Auto approach could resume printing when bed is cool and it's already too late to resume, thus causing double loss of filament.

For instance ASA on glass bed almost losses adhesion under 90-80 deg.C.

### CODE REVIEW

**Comment by thinkyhead** [comment]
I fundamentally disagree with allowing a machine to proceed with power-loss recovery without first getting user-confirmation, but that's just me. So I made the behavior optional and caused it to be disabled when the threshold is set to 0.

Values which can be edited and stored in EEPROM should always have an accompanying G-code. So I've added `M413 B` to configure this value.

See added commits for other changes.

**Comment by vovodroid** [comment]
> I fundamentally disagree with allowing a machine to proceed with power-loss recovery without first getting user-confirmation

But in this case time is against us - bed and  chamber are cooling, adhesion decreases, warping increases, and it could be situation "now or never", so it can't wait for user interaction.

By a way, shouldn't `#define EEPROM_VERSION` be increased as new member added?

**Comment by robherc** [comment]
I absolutely agree that the risk of breaking the printer itself can be unacceptably high when auto-restarting a print after a power loss, partly because we have no way of verifying how long the power was off.
However, I also understand the concern over lost adhesion & excessive warping...

So maybe a safe compromise, which could address both concerns would be to have the printer immediately re-enable bed & chamber heaters after a power loss (to halt further warping and loss of buildplate adhesion); then wait for user-intervention before continuing the print (so an actual human gets a chance to ensure that resuming the print would still be safe/sane & that nobody has any fingers in the way of the spicy bits).

**Comment by vovodroid** [comment]
> risk of breaking the printer itself can be unacceptably high when auto-restarting a print after a power loss

Why? What could happen to printer?

> to have the printer immediately re-enable bed & chamber heaters after a power loss

Great idea anyway!

**Comment by vovodroid** [comment]
See also  **[FR] Power Loss Recovery with UPS: auto resume printing if power was immediatly restored. #26604**

**Comment by trivalik** [comment]
If you know the nozzle/bed temperature is already near 0°C (best maybe below 40°C) or a certain range surround the set temperatur, then the power lost was longer than a few seconds.

BTW Gcode M1000 isn't documented somewhere.

**Comment by vovodroid** [comment]
> nozzle/bed temperature is already near 0°C (

Why zero? It shows room temperature.

**Comment by robherc** [comment]
> Why? What could happen to printer?

Assume power-loss happens near the end on a near-max-build-volume print & the printed part has come loose from the bed & warped into the path of the nozzle. Now moving the nozzle to print the next layer causes a collision with a (potentially extremely) solid part, which then collides with a section of the enclosure &/or printer frame, and if your motors don't stall first, whatever is most fragile breaks (likely some part of the print head).

There are other scenarios that could easily end in small broken parts (esp. bed probes), or a hopeless gooey mess being stuck to the nozzle...but I figured that would be the easiest one to illustrate the intent.

**Comment by vovodroid** [comment]
> & the printed part has come loose from the bed & warped into the path of the nozzle.

That's exactly what this PR is intended to prevent - resume if bed is still hot (probably we can take into account chamber as well).

**Comment by vovodroid** [comment]
> & the printed part has come loose from the bed & warped into the path of the nozzle.
 
That's exactly what this PR is intended to prevent - only resume if bed is still hot (probably we can take into account chamber as well).

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -1743,19 +1743,20 @@
...
   //#define POWER_LOSS_RECOVERY
   #if ENABLED(POWER_LOSS_RECOVERY)
-    #define PLR_ENABLED_DEFAULT   false // Power Loss Recovery enabled by default. (Set with 'M413 Sn' & M500)
-    //#define BACKUP_POWER_SUPPLY       // Backup power / UPS to move the steppers on power loss
-    //#define POWER_LOSS_ZRAISE       2 // (mm) Z axis raise on resume (on power loss with UPS)
-    //#define POWER_LOSS_PIN         44 // Pin to detect power loss. Set to -1 to disable default pin on boards without module.
-    //#define POWER_LOSS_STATE     HIGH // State of pin indicating power loss
-    //#define POWER_LOSS_PULLUP         // Set pullup / pulldown as appropriate for your sensor
+    #define PLR_ENABLED_DEFAULT       false // Power Loss Recovery enabled by default. (Set with 'M413 Sn' & M500)
+    //#define PLR_BED_THRESHOLD BED_MAXTEMP // (°C) Skip user confirmation at or above this bed temperature (0 to disable)
```
Comment (vovodroid) [inline Marlin/Configuration_adv.h:1747]: @thinkyhead 
Actually BED_MAXTEMP also disables the feature. Was it intention?

**Pull request merged**